state/cassandra/cassandra.go

```diff
@@ -0,0 +1,304 @@
+		meta.consistency = val
+	}
+
+	if val, ok := metadata.Properties[table]; ok && val != "" {
+		meta.table = val
+	}
+
+	if val, ok := metadata.Properties[keyspace]; ok && val != "" {
+		meta.keyspace = val
+	}
+
+	if val, ok := metadata.Properties[username]; ok && val != "" {
+		meta.username = val
+	}
+
+	if val, ok := metadata.Properties[password]; ok && val != "" {
+		meta.password = val
+	}
+
+	if val, ok := metadata.Properties[protoVersion]; ok && val != "" {
+		p, err := strconv.ParseInt(val, 0, 32)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing protoVersion field: %s", err)
+		}
+		meta.protoVersion = int(p)
+	}
+
+	if val, ok := metadata.Properties[replicationFactor]; ok && val != "" {
+		r, err := strconv.ParseInt(val, 0, 32)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing replicationFactor field: %s", err)
+		}
+		meta.replicationFactor = int(r)
+	}
+
+	return &meta, nil
+}
+
+// Delete performs a delete operation
+func (c *Cassandra) Delete(req *state.DeleteRequest) error {
+	query := fmt.Sprintf("DELETE FROM %s WHERE key='%s'", c.table, req.Key)
+	err := c.session.Query(query).Exec()
+	return err
+}
+
+// BulkDelete performs a bulk delete operation
+func (c *Cassandra) BulkDelete(req []state.DeleteRequest) error {
+	for _, re := range req {
+		err := c.Delete(&re)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Get retrieves state from cassandra with a key
+func (c *Cassandra) Get(req *state.GetRequest) (*state.GetResponse, error) {
+	query := fmt.Sprintf("SELECT value FROM %s WHERE key='%s'", c.table, req.Key)
+
+	session := c.session
+
+	if req.Options.Consistency == state.Strong {
+		sess, err := c.createSession(gocql.All)
+		if err != nil {
+			return nil, err
+		}
+		defer sess.Close()
+		session = sess
+	} else if req.Options.Consistency == state.Eventual {
+		sess, err := c.createSession(gocql.One)
+		if err != nil {
+			return nil, err
+		}
+		defer sess.Close()
+		session = sess
+	}
+
+	results, err := session.Query(query).Iter().SliceMap()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(results) == 0 {
+		return &state.GetResponse{}, nil
+	}
+	return &state.GetResponse{
+		Data: results[0]["value"].([]byte),
+	}, nil
+}
+
+// Set saves state into cassandra
+func (c *Cassandra) Set(req *state.SetRequest) error {
+	var bt []byte
+	b, ok := req.Value.([]byte)
+	if ok {
+		bt = b
+	} else {
+		bt, _ = jsoniter.ConfigFastest.Marshal(req.Value)
+	}
+
+	query := fmt.Sprintf("INSERT INTO %s (key, value) VALUES (?, ?)", c.table)
+
+	session := c.session
+
+	if req.Options.Consistency == state.Strong {
+		sess, err := c.createSession(gocql.Quorum)
+		if err != nil {
+			return err
+		}
+		defer sess.Close()
+		session = sess
+	} else if req.Options.Consistency == state.Eventual {
+		sess, err := c.createSession(gocql.Any)
+		if err != nil {
+			return err
+		}
+		defer sess.Close()
+		session = sess
+	}
+
+	err := session.Query(query, req.Key, bt).Exec()
+	return err
```

Comment:
I would just return session.Query(x) without the assignment.
```
return session.Query(query, req.Key, bt).Exec()
```

Reply:
Alright...

Reply:
Fixed with commit